src/server/scripts/Pet/pet_hunter.cpp

```diff
@@ -95,20 +95,20 @@ class npc_pet_hunter_snake_trap : public CreatureScript
                     Unit* summoner = me->ToTempSummon()->GetSummonerUnit();
 
                     std::vector<Unit*> targets;
-                    for (std::pair<ObjectGuid const, PvPCombatReference*> const& pair : summoner->GetCombatManager().GetPvPCombatRefs())
+
+                    auto check = [&](auto pair) mutable
```

Comment:
Then remove `const` from what Shauren said, and just call `check(second)`?

Reply:
Now should be good I think?